File: C++/c++/Lab date 21-09-2023/ShapePaint/Ellipse.cpp
#include "Ellipse.h"
#include <iostream>
using namespace std;
Ellipse::Ellipse()
{
}
Ellipse::Ellipse(Point p1(int xx,int yy))
{
	this->p=p1;
	this->xx=xx;
	this->yy=yy;
}
void Ellipse :: area()
{
	cout<<"Area of Ellipse: "<<p1.xx*p1.yy*3.142<<endl;
}

File: .NET/Day4/GDILib/GraphicNew/Ellipse.cs

namespace AWT.Draw;
using Hardware;

public class Ellipse:Shape,IPrinter{
    public Point Startpoint{get;set;}
    public Point Endpoint{get;set;}
    public Ellipse():base()
    {
        this.Startpoint = new Point(0,0);
        this.Endpoint = new Point(0,0);
    }
    public Ellipse(Point p1,Point p2,int w,String c):base(w,c)
    {
        this.Startpoint = p1;
        this.Endpoint = p2;
    }
    public override String ToString()
    {
        return base.ToString()+ " " +this.Startpoint+" "+this.Endpoint;
    }
    public override void Draw()
    {
        Console.WriteLine("Drawing is Ellipse...");
        Console.WriteLine(this);
    }
    void IPrinter.Print()
    {
        Console.WriteLine("printing Ellipse");
        Console.WriteLine(this);
    }
}